### DISCUSSION TITLE
Public Properties

### DISCUSSION THREAD

**Discussion by mikeabackhouse** [salesforce/apex-mockery#54] [Q&A] [answered]
Hi,
How do you handle mocked classes' public attributes/properties when using this framework? 
Aware that one option is to refactor into getters and setters, but just checking if there is an alternative without doing this across the codebase. 
Enjoying current use - thanks for building!

**Comment by LudoMeurillon** [comment, marked as answer]
We don't handle mocked public attributes/properties directly. 
The framework is focused on testing decoupled apex classes that use an apex interface on top of the collaborator implementation and stub its behavior using only the interface layer.

This means that if you have a dependency to another class public attribute that you want to mock then you should either:

- introduce getters methods, setters methods or both and mock its results with apex mockery
- create test that involves multiple dependant classes at a upper level in your design

If you can, feel free to share code samples so that we can discuss on your concrete use case

**Reply by mikeabackhouse** [reply]
I've gone the getters/setters route, and this is working well with a small refactor
